repo/gromacs/application.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2022-2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import os
+from ramble.appkit import *
+from ramble.expander import Expander
+
+
+class Gromacs(SpackApplication):
+    '''Define a Gromacs application'''
+    name = 'gromacs'
+
+    maintainers('douglasjacobsen')
+
+    tags('molecular-dynamics')
+
+    default_compiler('gcc9', spack_spec='gcc@9.3.0')
+    software_spec('impi2018', spack_spec='intel-mpi@2018.4.274')
+    software_spec('gromacs', spack_spec='gromacs@2020.5', compiler='gcc9')
+
+    executable('pre-process', 'gmx_mpi grompp ' +
+               '-f {input_path}/{type}.mdp ' +
+               '-c {input_path}/conf.gro ' +
+               '-p {input_path}/topol.top ' +
+               '-o exp_input.tpr', use_mpi=False)
+    executable('execute-gen', 'gmx_mpi mdrun -notunepme -dlb yes ' +
+               '-v -resethway -noconfout -nsteps 4000 ' +
+               '-s exp_input.tpr', use_mpi=True)
+    executable('execute', 'gmx_mpi mdrun -notunepme -dlb yes ' +
+               '-v -resethway -noconfout -nsteps 4000 ' +
+               '-s {input_path}', use_mpi=True)
+
```

Comment:
These invocations will need to be updated, but more extensive changes are probably best done in follow-up.

For now, I suggest removing the `-dlb yes` since this forces the load balancer on which is not compatible with certain features and might actually cause issues with the most recent release.

```suggestion
    executable('execute-nsteps-gen', 'gmx_mpi mdrun -notunepme ' +
               '-v -resethway -noconfout -nsteps 4000 ' +
               '-s exp_input.tpr', use_mpi=True)
    executable('execute-nsteps', 'gmx_mpi mdrun -notunepme ' +
               '-v -resethway -noconfout -nsteps 4000 ' +
               '-s {input_path}', use_mpi=True)
```

Reply:
To follow up on that, should we build in flexibility or keep the script features minimal (with respect to types of application invocation that may be tunable) for now?